pallets/merkle/src/lib.rs

```diff
@@ -201,7 +201,7 @@ pub mod pallet {
 	/// The map of verifying keys for each backend
 	#[pallet::storage]
 	#[pallet::getter(fn verifying_keys)]
-	pub type VerifyingKeys<T: Config> = StorageMap<_, Blake2_128Concat, Backend, Option<Vec<u8>>, ValueQuery>;
+	pub type VerifyingKeys<T: Config> = StorageMap<_, Blake2_128Concat, T::TreeId, Option<Vec<u8>>, ValueQuery>;
```

Comment:
I think a better way to do this would be to integrate a T::VerifyingKeyId and map TreeIds to VerifyingKeyIds and then map VerifyingKeyIds to Vec<u8>. This way trees can reuse verifying keys with minimal storage impact.